src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+    }
+
+    form.appendChild(label);
+  });
+
+  form.appendChild(button);
+  form.addEventListener('submit', addUser);
+  document.querySelector('table').after(form);
+}
+
+function selectUser() {
+  let selectedRow;
+
+  document.addEventListener('click', (e) => {
+    const currentRow = e.target.closest('tbody tr');
+
+    if (currentRow && currentRow !== selectedRow) {
+      if (selectedRow) {
+        selectedRow.classList.remove('active');
+      }
+      currentRow.classList.add('active');
+      selectedRow = currentRow;
+    } else if (!currentRow) {
+      if (selectedRow) {
+        selectedRow.classList.remove('active');
+      }
+    }
+  });
+}
+
+function sortLogic(headers, tbody) {
+  const sortOrder = {};
+
+  headers.forEach((header, index) => {
+    sortOrder[index] = true;
+
+    header.addEventListener('click', () => {
+      headers.forEach((_, i) => {
+        if (i !== index) {
+          sortOrder[i] = true;
+        }
+      });
+
+      const isAscending = sortOrder[index];
+
+      sortColumn(index, isAscending);
+      sortOrder[index] = !isAscending;
+    });
+  });
+
+  const sortColumn = (indexOfColumn, isAscending) => {
+    const rows = [...document.querySelectorAll('table tbody tr')];
+
+    rows.sort((a, b) => {
+      const aCell = a.children[indexOfColumn].textContent.trim();
+      const bCell = b.children[indexOfColumn].textContent.trim();
+
+      const aValue = parseFloat(aCell.replace(/[$,]/g, '')) || aCell;
+      const bValue = parseFloat(bCell.replace(/[$,]/g, '')) || bCell;
+
+      let result;
+
+      if (!isNaN(aValue) && !isNaN(bValue)) {
+        result = aValue - bValue;
+      } else {
+        result = aValue.localeCompare(bValue);
+      }
+
+      return isAscending ? result : -result;
+    });
+
+    rows.forEach((row) => {
+      tbody.appendChild(row);
+    });
+  };
+}
+
+function getFormData(form) {
+  const data = new FormData(form);
+  const nameInput = data.get('name');
+  const position = data.get('position');
+  const office = data.get('office');
+  const age = data.get('age');
+  const salary = +data.get('salary');
+
+  return {
+    name: nameInput,
+    position,
+    office,
+    age,
+    salary: `$${salary.toLocaleString('en-US')}`,
+  };
+}
+
+function addUser(e) {
+  e.preventDefault();
+
+  const form = document.querySelector('.new-employee-form');
+  const tbody = document.querySelector('tbody');
+
+  const user = getFormData(form);
+
+  if (user.name.length < 4 || !user.name.trim().length) {
+    createNotification(
+      'error',
+      'Fix your name',
+      'The name must consist of at least 4 letters.',
+    );
+
+    return;
+  }
+
+  if (!user.position.trim().length) {
+    createNotification(
+      'error',
+      'Fix your position',
+      'Position field can not be empty.',
+    );
+
+    return;
+  }
+
+  if (user.office === null) {
+    createNotification(
+      'error',
+      'Fix your office',
+      'Ofice field can not be empty.',
```

Comment:
There is a typo in the notification message: 'Ofice' should be corrected to 'Office'.